build_tools/travis/test_script.sh

```diff
@@ -67,3 +59,18 @@ fi
 if [[ "$SKIP_TESTS" != "true" ]]; then
     run_tests
 fi
+
+if [[ "$CHECK_PYTEST_SOFT_DEPENDENCY" == "true" ]]; then
+    conda remove -y py pytest || pip uninstall -y py pytest
+    if [[ "$COVERAGE" == "true" ]]; then
+        # Need to append the coverage to the existing .coverage generated by
+        # running the tests
+        CMD="coverage run --append"
```

Comment:
I'm not sure what the point of this is. The union of the coverage of running the tests with pytest and running the estimator checks without pytest?

Reply:
> I'm not sure what the point of this is. The union of the coverage of running the tests with pytest and running the estimator checks without pytest?

Yes. This way you can see that `run_tests_without_pytest` is covered in estimator_checks.py. test_script.sh is simpler without this so I can definitely remove it. With hindsight I probably should not have done it, it took me way too much time to get to work and that was only so that codecov would be green ...

Reply:
ok, fine by me. I'm good with merging either way.